main.js

```diff
@@ -45,79 +123,244 @@ function longestWord(sen) {
+      j++;
+    }
+  }
+
+  //this loop to store same longest length words into result array
+  let ind = 0;
+  for (let i = 0; i < ans.length; i++) {
+    if (ans[i].length === maxi) {
+      result[ind] = ans[i];
+      ind++;
+    }
+  }
+
+  //if array contain only one longest string then it return string
+  if (result.length == 1) {
+    return result[0];
+  }
+  //otherwise it returns result array
+  return result;
+}
 
 /* 8. Split an array into chunked arrays of a specific length
  * ex. chunkArray([1, 2, 3, 4, 5, 6, 7], 3) === [[1, 2, 3],[4, 5, 6],[7]]
  * ex. chunkArray([1, 2, 3, 4, 5, 6, 7], 2) === [[1, 2],[3, 4],[5, 6],[7]]
  */
-function chunkArray(arr, len) {}
-
-
+function chunkArray(arr, len) {
+  let ans = new Array();
+  console.log(arr.length);
+  for (let i = 0; i < arr.length; i++) {
+    let dummy = [];
+    while (dummy.length < len && i < arr.length) {
+      dummy.push(arr[i]);
+      i++;
+    }
+    ans.push(dummy);
+    i = i - 1;
+  }
+  return ans;
+}
 
 /* 9. Take an array of arrays and flatten to a single array
  * ex. flattenArray([[1, 2], [3, 4], [5, 6], [7]]) should return [1, 2, 3, 4, 5, 6, 7]
  */
-function flattenArray(arrays) {}
-
-
+function flattenArray(arrays) {
+  let result = new Array();
+  for (let i = 0; i < arrays.length; i++) {
+    for (let j = 0; j < arrays[i].length; j++) {
+      result.push(arrays[i][j]);
+    }
+  }
+  return result;
+}
 
 /* 10. Return true if anagram and false if not
  * ex. isAnagram('elbow', 'below') === true
  */
-function isAnagram(str1, str2) {}
-
-
+function isAnagram(str1, str2) {
+  let arr1 = new Array();
+  for (let i in str1) {
+    arr1.push(str1[i]);
+  }
+  arr1.sort();
+  console.log(arr1);
+  let arr2 = new Array();
+  for (let j in str2) {
+    arr2.push(str2[j]);
+  }
+  arr2.sort();
+  console.log(arr2);
+  if (arr1.length !== arr2.length) {
+    return "false";
+  } else {
+    for (let i = 0; i < arr1.length; i++) {
+      if (arr1[i] !== arr2[i]) {
+        return "false";
+      }
+    }
+  }
+  return "true";
+}
 
 /* 11. Change every letter of the string to the one that follows it (circularly), and capitalize the vowels
  * ex. 'hello there' === 'Ifmmp UIfsf'
  */
-function letterChanges(str) {}
-
-
+function letterChanges(str) {
+  let ans = "";
+  for (let i = 0; i < str.length; i++) {
+    if (str[i] === " ") {
+      ans = ans + " ";
+    } else {
+      let result = String.fromCharCode(str.charCodeAt(i) + 1);
+      if (
+        result === "a" ||
+        result === "e" ||
+        result === "o" ||
+        result === "i" ||
+        result === "u"
+      ) {
+        ans = ans + result.toUpperCase();
+      } else {
+        ans = ans + result;
+      }
+    }
+  }
+  return ans;
+}
 
 /* 12. Return a sum of all parameters entered regardless of the amount of numbers - NO ARRAYS
  * ex. addAll(2,5,6,7) === 20
  */
-function addAll() {}
-
-
+function addAll() {
+  let ans = 0;
+  for (let i = 0; i < arguments.length; i++) {
```

Comment:
learn about th hugher-order array method `reduce`, and combine it with spread operator

Reply:
Done